aws/cloudformation/standalone/gen_ai_curriculum/deploy_gen_ai_curriculum_stack

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env ruby
+require 'aws-sdk-applicationautoscaling'
+require 'colorize'
+require 'optparse'
+require 'open3'
+require 'pycall'
+require_relative './config'
+
+def main
+  options = {}
+  OptionParser.new do |opts|
+    opts.banner = "Usage: ./deploy_gen_ai_curriculum_stack [options]"
+
+    opts.on("-e", "--environment ENVIRONMENT", String, "Stack environment (prod or test)") do |v|
+      options[:environment] = v
+    end
+
+    opts.on("-d", "--dry-run [FLAG]", TrueClass, "Dry run (create template only but don't deploy)") do |v|
+      options[:dry_run] = v.nil? ? true : v
+    end
+
+    opts.on("-v", "--image-uri-version", Numeric, "HuggingFace Image URI version") do |v|
+      options[:image_uri_version] = v
+    end
+
+    opts.on("-i", "--image-uri IMAGE_URI", String, "HuggingFace Image URI to use") do |v|
+      options[:image_uri] = v
+    end
+
+    opts.on("-t", "--huggingface-token HUGGINGFACE_TOKEN", String, "HuggingFace Access Token") do |v|
```

Comment:
Currently this script requires a huggingface access token as a parameter. I've just been testing with my personal one. These tokens aren't _super_ secret since they're passed as an env var to the models, but we should at least not be hardcoding a token in our repo. Should we be pulling this from AWS Secrets and create an org-wide token?

Reply:
It looks like they support read only tokens too, it'd be good to use one of those in our final setup... assuming this token is for downloading models?

Reply:
Yeah good call, this is only used for downloading models so a read token should be sufficient.

Reply:
Moved this into secrets manager